src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -208,6 +208,14 @@ private void InnerReleaseHandle()
             }
         }
 
+        private void UnblockSocket(InnerSafeCloseSocket innerSocket)
+        {
+            if ((AsyncContext == null || !AsyncContext.GetNonBlocking()) && innerSocket != null && !_underlyingHandleNonBlocking && !innerSocket.IsClosed && !innerSocket.IsInvalid)
```

Comment:
I think we are doing all these tests to try to minimize the impact of this change. I.e. we want to only do the shutdown call in the case where we know it's hanging today. Correct?

I think it's fine to be cautious here, but we should add a comment explaining that these checks are basically just to be cautious (unless that's not true for some of them). If/when we revisit the socket close logic (and I hope we will do this at some point in 3.0), we will want to revisit this, and a comment will help explain the rationale behind this.

Reply:
yes. I can add comment with explanation. For Async/non-blocking we have no problem being stuck in system call.